server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -632,6 +635,16 @@ private ValuesSourceRegistry registerAggregations(List<SearchPlugin> plugins) {
                 .setAggregatorRegistrar(CompositeAggregationBuilder::registerAggregators),
             builder
         );
+        if (IndexSettings.isTimeSeriesModeEnabled()) {
+            registerAggregation(
+                new AggregationSpec(
+                    TimeSeriesAggregationBuilder.NAME,
+                    TimeSeriesAggregationBuilder::new,
+                    TimeSeriesAggregationBuilder.PARSER
+                ).addResultReader(InternalTimeSeries::new),
+                builder
+            );
```

Comment:
Note for anyone else scanning this PR - this skips registering the agg if the feature flag is not enabled. The agg is still registered regardless of the index's mode.